### PR TITLE
Hdrp decals htile buffer

### PR DESCRIPTION
**Pull request by Paulm-Unity**
### Purpose of this PR
Fixes decal HTile on Metal

---
### Release Notes
Please add any useful notes about the feature/fix that might be helpful for others.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=master&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?

**Halo Effect**: None, Low, Medium, High?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 1]
```diff
@@ -1,4146 +1,4153 @@
...
-                        targetId = hdCamera.xr.renderTarget;
-
-                    // Add render request
-                    var request = new RenderRequest
-                    {
-                        hdCamera = hdCamera,
-                        cullingResults = cullingResults,
-                        target = new RenderRequest.Target
-                        {
-
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: why the entire file is touch? if you haven't done any change, could you take the version from master and override your local one?

Comment (Paulm-Unity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs]: There is a change on line 336. No idea why github tagged the whole file as changed.

**Comment by TomasKiniulis** [comment]
Checking Windows editor currently and decals are not displayed at all on this branch

**Comment by TomasKiniulis** [comment]
> Checking Windows editor currently and decals are not displayed at all on this branch

Also couldn't get this working on Mac with the following warning `Metal: Fragment shader missing buffer binding at index 3 (_DecalPropertyMaskBuffer / HDRP/Decal)`

This is fixed now, but on Windows decals are still not displayed.

**Comment by TomasKiniulis** [comment]
Not encountering any issues anymore on Mac and Windows. Waiting for answers from Xbox and PS4 team members for the PR to be tested on these platforms

**Comment by sebastienlagarde** [comment]
Note: @Paulm-Unity I have try to regenerate th eHDRenderPipeline.cs file as the fact it touch all the files worry me. But this hasn't work. Still there is not only one line change there, there is severals change in the HDRenderPipeline.cs

I wait for the QA on others platformfor approval.

**Comment by sebastienlagarde** [comment]
hmm... Lauching locally I have no decals that appear on windows. @Paulm-Unity could you check? 
take a look at the commit of the update HDRenderPipeline.cs also just in case. It is suppose to be the same but we don't know.

**Comment by TomasKiniulis** [comment]
> I wait for the QA on others platformfor approval.

Unfortunately, neither testers of PS4 and Xbox teams were able to check it today. Will poke the guys on monday

**Comment by sebastienlagarde** [comment]
@Paulm-Unity I confirm that Yamato show that decal aren't working:
https://yamato-artifactviewer.prd.cds.internal.unity3d.com/9dd441bc-5b0f-40a4-bd2f-e54687682b9a%2Flogs%2FTestProjects%2FHDRP_Tests%2Fupm-ci~%2Ftest-results/TestReportV1.html

Decal don't appear...

You can see yamato build for this branch here:
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/hdrp-decals-htile-buffer/.yamato%252Fupm-ci-projects.yml%2523all_project_ci/493436/job/pipeline

Select windows play mode (red in the list):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/hdrp-decals-htile-buffer/.yamato%252Fupm-ci-hdrp.yml%2523HDRP_Windows64_playmode_DX11_trunk/493424/job/pipeline

Click on artifact and extend Logs:
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/hdrp-decals-htile-buffer/.yamato%252Fupm-ci-hdrp.yml%2523HDRP_Windows64_playmode_DX11_trunk/493424/job/artifacts

until you get the report v1
https://yamato-artifactviewer.prd.cds.internal.unity3d.com/9dd441bc-5b0f-40a4-bd2f-e54687682b9a%2Flogs%2FTestProjects%2FHDRP_Tests%2Fupm-ci~%2Ftest-results/TestReportV1.html

To relaunch yamato, select Run all test project and platform, and click Queue:
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/HDRP%252Ffix-reproj-exposure/.yamato%252Fupm-ci-projects.yml%2523all_project_ci

**Comment by TomasKiniulis** [comment]
I've rechecked the HDRP_Tests project and also get disappearing decals, but newly created projects don't seem to be affected by this and decals work correctly there. 

Unfortunately, neither Xbox or PS4 projects launched crashing on the start (PS4 log looks like this: [ps4-crash.txt](https://github.com/Unity-Technologies/ScriptableRenderPipeline/files/3640979/ps4-crash.txt)).
And switching to PS4 platform results in a list of the following errors:
`Start importing Packages/com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/Shaders/Denoising/DiffuseDenoiser.compute using Guid(b4ed2382141619f40af1f743a84ccaea) Importer(-1,)
Shader error in 'DiffuseDenoiser.compute': Program 'BilateralFilterColor', declaration is missing a variable name at kernel BilateralFilterColor at DiffuseDenoiser.compute(88) (on ps4)
Shader error in 'DiffuseDenoiser.compute': Program 'BilateralFilterSingle', declaration is missing a variable name at kernel BilateralFilterSingle at DiffuseDenoiser.compute(88) (on ps4)
Shader error in 'DiffuseDenoiser.compute': Program 'GatherColor', declaration is missing a variable name at kernel GatherColor at DiffuseDenoiser.compute(88) (on ps4)
Shader error in 'DiffuseDenoiser.compute': Program 'GatherSingle', declaration is missing a variable name at kernel GatherSingle at DiffuseDenoiser.compute(88) (on ps4)`

**Comment by TomasKiniulis** [comment]
@Paulm-Unity Adrian from Xbox team investigated why GPU hangs on start. He told there is an issue in ShaderPassDecal.hlsl PLATFORM_SUPPORTS_WAVE_INTRINSICS block. Although not too sure what exactly is the cause

**Comment by sebastienlagarde** [comment]
@TomasKiniulis Can you check again on windows, Xbox, PS4 please?

**Comment by sebastienlagarde** [comment]
duplicate of https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/4752

**Pull request closed without merging**